flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamTwoInputSelectableProcessor.java

```diff
@@ -195,14 +221,12 @@ public void cleanup() throws Exception {
 		}
 	}
 
-	private int selectNextReadingInputIndex()
-		throws InterruptedException, ExecutionException, IOException {
-
-		int readingInputIndex;
-		while ((readingInputIndex = inputSelection.fairSelectNextIndexOutOf2(availableInputsMask, lastReadInputIndex)) == -1) {
-			if (!waitForAvailableInput(inputSelection)) {
-				return -1;
-			}
+	private int selectNextReadingInputIndex() throws IOException {
+		updateAvailability();
```

Comment:
Looks weird to me to have this check (and maybe also the one on the next line) in this method. I would call this as part of `processInput` instead, e.g. before this method is called. I know it does the same, but somehow it feels like the check belong into the caller method and not down here.

Reply:
Hmmm, I would lightly disagree. `updateAvailability()` is (I think) only relevant for `selectNextReadingInputIndex()` so it should be tight to it in case of:
- someones refactor the code and moves `selectNextReadingInputIndex` somewhere else
- someones adds another code path to `selectNextReadingInputIndex`